parsl/dataflow/dflow.py

```diff
@@ -190,6 +192,71 @@ def __init__(self, config=Config()):
 
         atexit.register(self.atexit_cleanup)
 
+    def fast_abort(self):
+        """
+        Trigger a fast abort. this method should return fast and be
+        threadsafe, so that it is easy to call from anywhere inside or
+        outside of the DFK.
+
+        This will be triggered inside the DFK by an app failing when
+        non-lazy failures (rename that...) are enabled; but we should
+        expect other places to want to trigger this behaviour. Examples:
+          i) ctrl-C should not cause the dfk cleanup to wait for a long
+             time
+          ii) sometimes executors will fail so badly they want to shut
+              down the workflow (@yadudoc)
+
+        The fast abort flag can only move from False to True. Once
+        a fast abort has started, it cannot be cancelled.
+        What does "fast abort" mean? It doesn't mean that the DFK
+        will shutdown absolutely as fast as it can. It means that: (?)
+        i) no more tasks will be launched.
+        ii) DFK cleanup will not wait for appfutures to complete/tasks
+            to finish. DONE
+        iii) what will happen with executor cleanup? This can take a
+             moderate amount of time. But I think we *do* want to do
+             enough cleanup to kill running tasks right now (rather than
+             either abandoning them to keep running, or waiting for
+             them to end) TODO
+        iv) checkpointing will not happen TODO / done in cleanup. Is checkpointing "expensive"? if so we should not do it. if it is "cheap" we should do it.
+        v) fast abort *does not* exit the process: it means that the DFK
+           should clean itself up fairly fast, and that parsl operations
+           involving this DFK should throw an exception if they cannot
+           be completed immediately (for example, all unset AppFutures
+           should get an exception status immediately)
+        """
+        logger.warn("Fast-abort activated")
+        self._fast_abort=True
+        # what needs to happen here in addition to the rest of the
+        # DFK failing fast as it notices?
+
+    def set_termination_handler(self):
```

Comment:
Where is the idea to call this from?

Reply:
For now, at the start, roughtly where you might call set stream logger.

Adding signal handlers by default is a bit of an awkward thing to do for a library (in the same way that setting logging by default is).

In the case of a standalone workflow, it might make sense to do this automatically at the start of DFK initialisation, but I'm not sure when running in a notebook.

Reply:
Yikes. I see your point, but I also think this could be a bit obtuse to explain to new users, and something people could easily forget to add to their scripts. I'd estimate that in 99% of the cases we encounter (in standalone scripts, at least), people will have wanted to set these signal handlers. Maybe, if we're not willing to add signal handlers by default, we need to undo the automatic call to `wait_for_current_tasks` we added in #610, and instead rely on users to call that themselves. We could add a shortcut `parsl.wait()` or similar so the user wouldn't have to interact with the DFK directly.